### PR TITLE
Testing framework setup

### PR DESCRIPTION
**Pull request by vsima**
- add robotium and test config for build.gradle
- add test for login screen
- add signup activity test
- add article activity test
- add reading activity test
- add topics activity test
- add tests for user metrics activity
- add tests for about activity
- add tests for feedback activity
- add exercise activity test
- add settings activity test
- fix teardown bug
- update readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,13 @@ Clone the sourcecode.
     cd kanshu-android
     ./gradlew assembleRelease
 
+#### Testing
+Make sure you have an emulator running or preferribly a device connected via usb and unlocked
+
+    ./gradlew connectedTest
```
Comment (intrications) [inline README.md]: Should this be `./gradlew connectedAndroidTest`?

**Comment by intrications** [comment]
Just ran `./gradlew connectedAndroidTest` with a Galaxy S5 and got:

```
:app:connectedAndroidTest

com.kanshu.kanshu.test.activity.ExerciseActivityTest > testExerciseScreenAppeared[SM-G900F - 4.4.2] FAILED 
Test failed to run to completion. Reason: 'Instrumentation run failed due to 'java.lang.OutOfMemoryError''. Check device logcat for details
Tests on SM-G900F - 4.4.2 failed: Instrumentation run failed due to 'java.lang.OutOfMemoryError'
:app:connectedAndroidTest FAILED
```

Am I doing anything wrong? I wouldn't expect the S5 to be too limited in memory.

**Comment by intrications** [comment]
However it worked fine on a Nexus 6.

Thanks for this. I'll probably be able to merge this once you take a look at my queries.

**Comment by vsima** [comment]
Hmm I normally run ./gradlew connectedCheck to run the tests. I'll have to
see If its been deprecated. The tests work ok on my nexus 5 and on the
latest travis emulator image. Interested to see what the issue is with the
s5.

**Comment by vsima** [comment]
Oh . just noticed the typo in the readme. Try ./gradlew clean
connectedCheck

**Comment by intrications** [comment]
Yep, I just checked ./gradlew tasks and saw connectedCheck. That works.

**Comment by vsima** [comment]
ok, i updated the README. it should be ready to go.

**Comment by intrications** [comment]
Thanks. Merged.

**Pull request merged**